modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/BaseSystemObjectRelatedModelsProviderTestCase.java

```diff
@@ -0,0 +1,496 @@
+			0, _objectRelatedModelsProvider,
+			_objectRelationship.getObjectRelationshipId(), primaryKeys[2],
+			StringUtil.randomString());
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			1, _objectRelatedModelsProvider,
+			_objectRelationship.getObjectRelationshipId(), primaryKeys[2],
+			String.valueOf(objectEntry3.getObjectEntryId()));
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			1, _objectRelatedModelsProvider,
+			_objectRelationship.getObjectRelationshipId(), primaryKeys[2],
+			"New");
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			2, _objectRelatedModelsProvider,
+			_objectRelationship.getObjectRelationshipId(), primaryKeys[2],
+			"Entry");
+
+		_objectRelationshipLocalService.deleteObjectRelationships(
+			_objectDefinition.getObjectDefinitionId());
+
+		Assert.assertNull(
+			_objectRelationshipLocalService.fetchObjectRelationship(
+				_objectRelationship.getObjectRelationshipId()));
+	}
+
+	protected long[] addSystemObjectEntry(int count) throws Exception {
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	protected void assertFailureNoSuchException(long primaryKey)
+		throws Exception {
+
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	protected void deleteSystemObjectEntry(long primaryKey) throws Exception {
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	protected Object fetchSystemObjectEntry(long primaryKey) {
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	protected ObjectDefinition getSystemObjectDefinition() throws Exception {
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	protected String getSystemObjectEntryName(long primaryKey)
+		throws Exception {
+
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
+	}
+
+	private void _addObjectRelationship(
+			ObjectDefinition objectDefinition1,
+			ObjectDefinition objectDefinition2, String deletionType,
+			String relationshipType)
+		throws Exception {
+
+		_objectRelationship =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				objectDefinition1.getObjectDefinitionId(),
+				objectDefinition2.getObjectDefinitionId(), 0, deletionType,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(), relationshipType);
+
+		if (!StringUtil.equals(
+				relationshipType,
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) {
+
+			_relationshipObjectField = _objectFieldLocalService.getObjectField(
+				_objectRelationship.getObjectFieldId2());
+		}
+
+		_objectRelatedModelsProvider =
+			_objectRelatedModelsProviderRegistry.getObjectRelatedModelsProvider(
+				objectDefinition2.getClassName(),
+				objectDefinition2.getCompanyId(), relationshipType);
+	}
```

Comment:
@pedro-oliveira446 why you didn't move this method to `ObjectRelationshipTestUtil`?

Reply:
Hey @carolmariaabb , this method populates some class variables like `_objectRelationship`, `_relationshipObjectField` and `_objectRelatedModelsProvider`. The only way I see is just moving the first block, I don't know if it's worth it, what do you think?

Reply:
@guilhermedcamacho we can leave it as it is without changing. Thanks!